From 0bada539aa5e10c9651b3d8f0793645c21a89e0f Mon Sep 17 00:00:00 2001
From: Dorian Stoll <dorian.stoll@tmsp.io>
Date: Sat, 22 Jul 2023 10:45:33 +0200
Subject: [PATCH] Use a custom key and certificate for Secure Boot signing

Signed-off-by: Dorian Stoll <dorian.stoll@tmsp.io>
---
 redhat/kernel.spec.template | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/redhat/kernel.spec.template b/redhat/kernel.spec.template
index b371a121..17603f2c 100644
--- a/redhat/kernel.spec.template
+++ b/redhat/kernel.spec.template
@@ -935,6 +935,7 @@ BuildRequires: system-sb-certs
 %ifarch x86_64 aarch64 riscv64
 BuildRequires: nss-tools
 BuildRequires: pesign >= 0.10-4
+BuildRequires: sbsigntools
 %endif
 %endif
 %endif
@@ -1018,6 +1019,13 @@ Source13: redhatsecureboot501.cer
 %define signing_key_filename kernel-signing-s390.cer
 %endif
 
+%ifarch x86_64 aarch64
+
+Source7001: MOK.key
+Source7002: MOK.crt
+
+%endif
+
 # pesign cert name is auto-discovered during build from secureboot_key_0,
 # see pesign_name_0 shell variable
 #
@@ -2698,7 +2706,7 @@ BuildKernel() {
 
     %ifarch x86_64 aarch64
     %{log_msg "Sign kernel image"}
-    %pesign -s -i $SignImage -o vmlinuz.signed -a %{secureboot_ca_0} -c %{secureboot_key_0} -n $pesign_name_0
+    sbsign --key %{SOURCE7001} --cert %{SOURCE7002} --output vmlinuz.signed $SignImage
     %endif
     %ifarch s390x ppc64le
     if [ -x /usr/bin/rpm-sign ]; then
-- 
2.43.0

